Made header, nav and body independent of COVID banner

parulmishra opened the pull request:
Made header, nav and body independent of COVID banner using antd components.

thegentlemanphysicist commented on react-frontend/src/components/Nav/breadcrumbs.js:
```diff
@@ -1,19 +1,26 @@
+   
 
   useEffect(() => {
+    console.log("props--",props);
```
Forgotten console log

thegentlemanphysicist commented on react-frontend/src/components/Nav/navbar.js:
```diff
@@ -1,7 +1,11 @@
 import React, { useState, useEffect } from 'react';
+import { Drawer, Menu } from 'antd';
+import '../../css/nav.css';
```
This file no longer exists, the project is not able to run due to this.

parulmishra replied:
Merge artifact. Removing this.

thegentlemanphysicist commented on react-frontend/src/components/Nav/breadcrumbs.js:
```diff
@@ -1,19 +1,26 @@
 } from '../StyleComponents/htmlTags';
 import { BreadcrumbContainer } from '../StyleComponents/pageContent';
-const BreadCrumbs = () => {
+import {Col} from "antd";
```
unused import

thegentlemanphysicist reviewed (changes requested):
Due to this branch importing a css file that no longer exists, the branch crashes when I try to run it. I commented on a few tiny things.  Once the branch is deployable, I can do a more in-depth  code review.

thegentlemanphysicist commented on react-frontend/src/components/StyleComponents/htmlTags.js:
```diff
@@ -146,6 +146,10 @@ export const NavBarHeaderLink = styled.a.attrs({
   line-height: 1;
   padding: 5px;
+  margin-left: 3rem;
+  @media screen and (max-width: 800px) {
```
I believe that this is also messing with the alignment of the logo and nav bar on small screens

thegentlemanphysicist commented on react-frontend/src/components/StyleComponents/htmlTags.js:
```diff
@@ -177,23 +181,15 @@ export const NavBarLink = styled(Link).attrs({
   display: flex;
   flex-direction: column;
   list-style: none;
+  margin-left: -75px;
```
This was set to zero in the refactor so that the NavBarContainer could control the left alignment of the page

parulmishra replied:
I don't understand where and how this was being used. But with my refactor (because this is the wild west and we are all gunslingers refactoring code as we want to) things have changed. This style component is not being used at all. I am reverting this change. Whoever wants to cleanup the code or reuse this component can feel free to do so. :)

thegentlemanphysicist commented on react-frontend/src/components/StyleComponents/htmlTags.js:
```diff
@@ -202,4 +198,7 @@ export const NavBarUl = styled.ul.attrs({
   @media screen and (min-width: 800px) {
     flex-direction: row;
   }
+  @media screen and (max-width: 800px) {
```
See previous comment.

parulmishra replied:
See previous comment.

thegentlemanphysicist reviewed (commented):
I've left a lot of comments on this one. I think the introduction of Menu and Menu.Item undid the work in ticket #193 .

mark-a-wilson reviewed (changes requested):
Thanks @parulmishra — this is a big improvement and resolves 1 of the 2 problems mentioned in #360, but I've noticed it also introduces these issues:

- [ ] The `<nav>` needs to have `top: 67px` so that it doesn't overlap the 2px yellow `border-bottom` on the `<header>`

- [ ] The `<li class="ant-menu-item ant-menu-item-only-child ant-menu-item-selected">` has a `border-bottom` style that is not part of the Design System. It should be removed so that `<li>`  items in the `<nav>` don't inherit that blue border-bottom style in any circumstances.

- [ ] The alignment of the BCGov log `<navBarHeaderLink>`, and the first item (Home) in the `<nav>` are now out of alignment and jumping around. This reintroduces  Issue #193 which Jon recently fixed in #377 😭. The behaviour we'd like is what you can observe by resizing a window on either our current test deployment or DevHub (The items stay 'locked' together).

- [ ] The Drawer pattern you chose for the mobile menu is kind of cool (slides in from the side), but the standard interaction I've seen on Gov apps (example DevHub) is a vertical dropdown show/hide. Ant Design's [Menu patterns](https://ant.design/components/menu/) might be more appropriate than the Drawer? But if too much work to swap, we can get away with the Drawer as you've implemented.

And (though this issue was not introduced by your PR but was present before) if we could also resolve:
- [ ] (the second problem mentioned in #360) the issue with the mobile nav menu overlapping the header and hamburger button (only occurs on mobile displays when the COVID banner is still present), that would be great!

🙏🙏🙏🙏🙏

parulmishra commented on react-frontend/src/components/StyleComponents/htmlTags.js:
```diff
@@ -117,7 +117,7 @@ export const HrefLinkStandaloneInternal = styled(Link).attrs({
   font-size: 16px;
   font-weight: bold;
-  margin-bottom: 16px;
+
```
This isn't the footer. This is used in the breadcrumb and other places.
To be honest, this change doesn't seem to be doing anything. I am reverting it back in any case.

thegentlemanphysicist replied:
ah, the way github had collapsed the text made me read that incorrectly

parulmishra commented on react-frontend/src/components/Nav/breadcrumbs.js:
```diff
@@ -44,12 +58,30 @@ const BreadCrumbs = () => {
   const crumbs = (
     <div>
-      <BreadcrumbUL>{listEntry}</BreadcrumbUL>
+      <BreadcrumbUL
```
I tried. It wasn't working.

parulmishra commented on react-frontend/src/components/Nav/covidbanner.js:
```diff
@@ -47,12 +48,14 @@ function CovidBanner() {
+                            props.test(bredCrub);
                           }}
                         >
                           <font color="white">X</font>
```
This link doesn't have any styling. It is taking some action based on the link onClick event. Why do we need to make it a styled component? 😕

mark-a-wilson commented:
Noticed another thing on the deployment:

- [ ] The `<ul>` with class  `.ant-menu-horizontal` has `border-bottom: 1px solid #f0f0f0`. Can you please remove this by adding `border: none`.

![border-bottom](https://user-images.githubusercontent.com/16946297/97896990-b2a4c580-1cea-11eb-8d17-eb7eb01609b8.png)

mark-a-wilson commented:
Another observation—forgive me, I know this is tedious, but...

Per the Design System, the `<header>` is supposed to always have `border-bottom: "2px solid rgb(252, 186, 25)"`. 

On small displays, it is good, but between `800px` and `1120px`, it is becoming thinner—only `1px`. It seems to fix itself at `1121px`. I presume this has something to do with the `position` of something, because `1121px` is when the text in the COVID banner wraps onto a single line. It just makes me think that perhaps the COVID banner and everything else aren't quite "independent" yet, if one is affecting the other?

![screen width 1120px](https://user-images.githubusercontent.com/16946297/97899407-1e3c6200-1cee-11eb-8f44-02c06238329e.png)

![screen width 1121px](https://user-images.githubusercontent.com/16946297/97899417-21cfe900-1cee-11eb-819f-fddb2f5a1a31.png)

thegentlemanphysicist commented:
This is still breaking a lot of alignment in the nav, can we have a conversation about this over teams?

mark-a-wilson commented on react-frontend/src/components/StyleComponents/nav.js:
```diff
@@ -168,3 +177,79 @@ export const SkipToMainContent = styled.a.attrs({
+    word-wrap: break-word;
+    background-color: #38598a;
+  }
+  height: 130px;
```
Can we remove the fixed `height` here?

![ant-drawer-body](https://user-images.githubusercontent.com/16946297/97904263-146a2d00-1cf5-11eb-8e62-461b4dcd133d.png)

parulmishra replied:
If I remove the height:
![image](https://user-images.githubusercontent.com/20617793/97915572-872fd400-1d06-11eb-9e2f-63effdc35e1b.png)

Increasing the height to 140px to ensure similar top and bottom.

mark-a-wilson reviewed (changes requested):
Hey @parulmishra -- apologies for all of these comments. But I'm feeling like the fundamental issue (#360) isn't getting addressed in this PR. 

Is there a way for us to achieve proper responsive layout in both states (COVID banner is present or dismissed) that avoids all uses of `position: fixed` and `top: xx` on different elements? 

In particular, I feel that a lot of these finicky styling issues are a result of: 

In **StyleComponents/nav.js**, the `top:` properties defined for different breakpoints on two elements, **id: 'navheader'** and **id: 'navbar'**.

![Screen Shot 2020-11-02 at 10 35 55 AM](https://user-images.githubusercontent.com/16946297/97905781-75930000-1cf7-11eb-8e44-53d9bb64cca9.png)
![Screen Shot 2020-11-02 at 10 36 09 AM](https://user-images.githubusercontent.com/16946297/97905785-775cc380-1cf7-11eb-9df9-f5a6c8665780.png)

thegentlemanphysicist commented on react-frontend/src/components/StyleComponents/htmlTags.js:
```diff
@@ -177,11 +184,49 @@ export const NavBarLink = styled(Link).attrs({
 
-export const NavBarLinkFirst = styled(NavBarLink).attrs({
-  className: 'navBarLinkFirst',
+export const NavBarLinkResource = styled(Link).attrs({
```
Having custom styling on each nav bar element is a red flag css code wise.  in the original pattern there was only 'special' styling on the first element.  The rest shared the same style

thegentlemanphysicist commented on react-frontend/src/components/StyleComponents/nav.js:
```diff
@@ -72,10 +72,11 @@ export const NavHamburgerButton = styled.div.attrs({
     cursor: pointer;
     display: none;
-    margin: 0 0 0 auto;
+    margin: 0 0 20px auto;
```
Changing left alignment is part of what is breaking alignment

thegentlemanphysicist reviewed (commented):
There several issues with the PR
- If the page is resized when the dropdown menu is open, it can no longer be dismissed and it's styling gets broken. (this is an actual bug)
- there is custom styling on each element in the nav bar menu. In my experience this is usually a red flag for styling patterns. In current dev, the only item with it's own styling is the first element. All others have the same spacing, padding, etc.
- The alignment of the logo and the navbar are no longer fixed see issue #193.  This applies to both large screens and the drop down nav bar.

The reason I want to have an 'in person' chat is because I'm having trouble understanding why the nav bar has been refactored in this way and how it's addressing issues with the covid banner.  I also feel like I haven't been as clear as I could be with certain general styling principles that I've been trying to implement.

parulmishra commented:
> Hey @parulmishra -- apologies for all of these comments. But I'm feeling like the fundamental issue (#360) isn't getting addressed in this PR.
> 
> Is there a way for us to achieve proper responsive layout in both states (COVID banner is present or dismissed) that avoids all uses of `position: fixed` and `top: xx` on different elements?
> 
> In particular, I feel that a lot of these finicky styling issues are a result of:
> 
> In **StyleComponents/nav.js**, the `top:` properties defined for different breakpoints on two elements, **id: 'navheader'** and **id: 'navbar'**.
> 
> ![Screen Shot 2020-11-02 at 10 35 55 AM](https://user-images.githubusercontent.com/16946297/97905781-75930000-1cf7-11eb-8e44-53d9bb64cca9.png)
> ![Screen Shot 2020-11-02 at 10 36 09 AM](https://user-images.githubusercontent.com/16946297/97905785-775cc380-1cf7-11eb-9df9-f5a6c8665780.png)

Thanks for the input @mark-a-wilson . I have tried multiple ways to do it without specifying an explicit top property but was unable to do so. Because the element's position is fixed, so it doesn't flow (as it would have). The only way to make it flowing (depending upon content / size) is to have a top and manipulate it according to screen size. I am open to suggestion here. Please feel free to create a new ticket and track this or feel free to pull in this branch locally and try out a fix. I really appreciate your suggestions here.

parulmishra commented:
> There several issues with the PR
> 
> * If the page is resized when the dropdown menu is open, it can no longer be dismissed and it's styling gets broken. (this is an actual bug)
> * there is custom styling on each element in the nav bar menu. In my experience this is usually a red flag for styling patterns. In current dev, the only item with it's own styling is the first element. All others have the same spacing, padding, etc.
> * The alignment of the logo and the navbar are no longer fixed see issue #193.  This applies to both large screens and the drop down nav bar.
> 
> The reason I want to have an 'in person' chat is because I'm having trouble understanding why the nav bar has been refactored in this way and how it's addressing issues with the covid banner. I also feel like I haven't been as clear as I could be with certain general styling principles that I've been trying to implement.

Thanks for the input @thegentlemanphysicist . 
1. Page resize - I will try to figure out a fix for this. Given that we are using drawers and menu antd components, it might not be as straightforward. What I am trying to understand is the actual usecase for this scenario? When will the user actually face this? Given the extreme corner case and an available workaround (refreshing the page), it seems like a low priority work item. 
3 - The navbar looks aligned to me. 😕  Can you please add some details here? a snapshot or dimensions or code of what it should be?
2 - I believe you have much more experience with styling, styling guidelines, css and react style components in general. I believe I have refactored it as it was needed. Moving styling down to a lower (more granular) level doesn't feel like an anti-pattern to me. Please feel free to pull in the branch and make changes if you feel like. :)

parulmishra commented:
Abandoning the PR
